examples/ota-provider-app/ota-provider-common/BdxOtaSender.cpp

```diff
@@ -82,13 +82,18 @@ void BdxOtaSender::HandleTransferSessionOutput(TransferSession::OutputEvent & ev
             // end of the transfer.
             sendFlags.Set(chip::Messaging::SendMessageFlags::kExpectResponse);
         }
-        VerifyOrReturn(mExchangeCtx != nullptr, ChipLogError(BDX, "%s: mExchangeCtx is null", __FUNCTION__));
+        VerifyOrReturn(mExchangeCtx != nullptr, ChipLogError(BDX, "mExchangeCtx is null"));
         err = mExchangeCtx->SendMessage(event.msgTypeData.ProtocolId, event.msgTypeData.MessageType, std::move(event.MsgData),
                                         sendFlags);
         if (err != CHIP_NO_ERROR)
         {
             ChipLogError(BDX, "SendMessage failed: %s", chip::ErrorStr(err));
```

Comment:
On send error here, shouldn't we close and null out `mExchangeCtx`?  Otherwise don't we leak it?

Or does something higher up the callstack handle that?  But then the fact that we are closing it in the next block unconditionally, whether the send succeeded or not, is wrong....

Reply:
Calling `Reset()` here, which closes the exchange and set it to `NULL`